Boxfuse: Specified key was too long; max key length is 767 bytes

<!--
- Please follow the issue template below for bug reports and feature requests.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

I created a fresh project using JHipster 4.14.1 version. I used MySQL in both production and development. No entities or edits to code were done.

When deploying it, I get this error `Specified key was too long; max key length is 767 bytes`

```
i-0f39649480c8e => 2018-03-07 13:18:36.547 Caused by: liquibase.exception.DatabaseException: Specified key was too long; max key length is 767 bytes [Failed SQL: ALTER TABLE incuswebapp.jhi_social_user_connection ADD UNIQUE (user_id, provider_id, provider_user_id)]

```


##### **Reproduce the error**

1. Generate a JHipster version using v4.14.1. Use MySQL for production and development.
2. Run `./mvnw package -Pprod -DskipTests`
3. Run `boxfuse run -env=prod`

##### **Related issues**

This Laravel issue seems relevant, have a look [here](https://stackoverflow.com/questions/23786359/laravel-migration-unique-key-is-too-long-even-if-specified).

##### **JHipster configuration**

INFO:

mohataher-pc:~/dev/myapp$ jhipster info


Using JHipster version installed locally in current project's node_modules
Executing jhipster:info
Options: 
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
incuswebapp@0.0.0 /home/mohataher/dev/incuswebapp4
└── generator-jhipster@4.14.1 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.incus.web"
    },
    "jhipsterVersion": "4.14.1",
    "baseName": "incuswebapp",
    "packageName": "com.incus.web",
    "packageFolder": "com/incus/web",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": true,
    "enableSwaggerCodegen": true,
    "jwtSecretKey": "replaced-by-jhipster-info",
    "clientFramework": "angularX",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": false
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>




</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_151"
Java(TM) SE Runtime Environment (build 1.8.0_151-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.151-b12, mixed mode)

git version 2.7.4

node: v9.4.0

npm: 5.6.0

yeoman: 2.0.0

yarn: 1.3.2

--- a/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
+++ b/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
@@ -242,13 +242,13 @@
             <column name="id" type="bigint" autoIncrement="${autoIncrement}">
                 <constraints primaryKey="true" nullable="false"/>
             </column>
-            <column name="user_id" type="varchar(255)">
+            <column name="user_id" type="varchar(191)">
                 <constraints nullable="false"/>
             </column>
-            <column name="provider_id" type="varchar(255)">
+            <column name="provider_id" type="varchar(191)">
                 <constraints nullable="false"/>
             </column>
-            <column name="provider_user_id" type="varchar(255)">
+            <column name="provider_user_id" type="varchar(191)">
                 <constraints nullable="false"/>
             </column>
             <column name="rank" type="bigint">

